### ISSUE TITLE
Sort jsdoc parameter suggestions by argument position 

### ISSUE THREAD

**Issue by mjbvz** [microsoft/TypeScript#20183]
From https://github.com/Microsoft/vscode/issues/35124

**TypeScript Version:**  2.7.0-dev.20171116

**Code**

For the code:

```ts
/**
 * @param |
 */
function foo(z, a) {}
```

Trigger completions at the `|` after `@param`

**Expected behavior:**
`z` is returned first, followed by `a`. The `sortText` can be used for this

**Actual behavior:**
`a` and `z` are returned with sort orders weights:

```
[Trace  - 11:33:07 PM] Sending request: completions (176). Response expected: yes. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/y.ts",
    "line": 2,
    "offset": 11,
    "includeExternalModuleExports": true
}
[Trace  - 11:33:07 PM] Response received: completions (176). Request took 2 ms. Success: true 
Result: [
    {
        "name": "a",
        "kind": "parameter",
        "kindModifiers": "",
        "sortText": "0"
    },
    {
        "name": "z",
        "kind": "parameter",
        "kindModifiers": "",
        "sortText": "0"
    }
]
```

**Comment by mhegazy** [comment]
PRs welcomed.

**Comment by bmsdave** [comment]
@mjbvz tell me how to reproduce this situation, plz.
Where can I see the trace that is specified in the issue description?
`[Trace  - 11:33:07 PM] Sending request ***`

**Comment by mjbvz** [comment]
1. In your vscode user settings, set `"typescript.tsserver.trace": "verbose"`.
2. Open output panel and go to the `TypeScript` section

**Comment by SimonGustavsson** [comment]
Is this still an issue? I tried this:
```ts
/** */
public test(b: number, a: number) {
    console.log(a);
    console.log(b);
}
```
 with the caret placed after `**` auto complete shows a completion of "JSDoc comment", hitting Return produces this, with the caret placed after `@param b`
```ts
/**
 * 
 * @param b 
 * @param a 
 */
public test(b: number, a: number) {
    console.log(a);
    console.log(b);
}
```

**Comment by mjbvz** [comment]
Yes. It is correct when using a template to generate the doc comment. This issue tracks the sort order when you trigger suggestions after `@param`.

**Comment by shonnungar** [comment]
Looking into this issue

**Comment by RafaelMeir4** [comment]
Hi @mjbvz, This is my first time contributing to an open-source project, and I’m really excited to start! I’ve read the issue and I’m planning to go through the contribution guidelines to make sure I follow the process correctly. I’d like to work on this issue if it’s still relevant.

Before I dive in, I wanted to ask:

Is this issue still a priority, or has it already been resolved in recent updates?
Could you point me to the folder or file where the related logic is implemented, so I can start exploring the codebase?
I’ve already checked the repository and have some ideas about where to start, but any pointers would help me focus my efforts.

Thank you in advance for your guidance, and I’m looking forward to contributing!

**Comment by ManishPrakkash** [comment]
Checking in to see if this is still open for a PR? I’d like to help out with the requested changes

**Comment by LeonxLJX** [comment]
I'd like to take this one (`Sort jsdoc parameter suggestions by argument position`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)

**Comment by LeonxLJX** [comment]
I'd like to take this one (`Sort jsdoc parameter suggestions by argument position`). I'll dig into the root cause and follow up with a PR shortly. (claiming via @LeonxLJX)

**Comment by LeonxLJX** [comment]
I'd like to take this one (`Sort jsdoc parameter suggestions by argument position`). I'll look into the root cause and follow up with a PR. (claiming via @LeonxLJX)
